WakaTime.py

```diff
@@ -103,6 +110,49 @@ def invoke(self, context, event):
         return context.window_manager.invoke_props_dialog(self)
 
 
+# Addon prefs
+class WakaTimePreferences(bpy.types.AddonPreferences):
+    bl_idname = __name__
+    _def_chars="<empty>"
+    _def_prefix="<empty>"
+    _def_postfix="<empty>"
+    if len(_default_chars) > 0:
+        _def_chars=_default_chars
+    if len(_default_prefix) > 0:
+        _def_prefix=_default_prefix
+    if len(_default_postfix) > 0:
+        _def_postfix=_default_postfix
+    always_overwrite_name: BoolProperty(
+        name = "Overwrite project-discovery with the name from below",
+        default = _default_always_overwrite_projectname,
+        description="WakaTime will guess the project-name (e.g. from the git-repo). Checking this box will overwrite this auto-discovered name (with the name according to the rules below).\n\nHint: when not working with git, the project's name will always be set according to the rules below.\n\nDefault: " + str(bool(_default_always_overwrite_projectname)))
+    use_project_folder: BoolProperty(
+        name = "Use folder-name as project-name",
+        default = _default_use_project_folder,
+        description="Will use the name of the folder/directory-name as the project-name.\n\nExample: if selected, filename 'birthday_project/test_01.blend' will result in project-name 'birthday_project'\n\nHint: if not activated, the blender-filename without the blend-extension is used.\n\nDefault: " + str(bool(_default_use_project_folder)))
+    truncate_trail: StringProperty(
+        name = "Cut trailing characters",
+        default = _default_chars,
+        description="With the project-name extracted (from folder- or filename), these trailing characters will be removed too.\n\nExample: filename 'birthday_01_test_02.blend' will result in project-name 'birthday_01_test'\n\nDefault: '" + _def_chars + "'")
```

Comment:
Don't concatenate strings when you can use formatting
```suggestion
        description = f"With the project-name extracted (from folder- or filename), these trailing characters will be removed too.\n\nExample: filename 'birthday_01_test_02.blend' will result in project-name 'birthday_01_test'\n\nDefault: '{_def_chars}'")
```

Reply:
Also, why the `description` uses `_def_chars`, while the `default = _default_chars`?

Reply:
Format will be fixed :+1: 

The `_def_chars` vs. `_default_chars` is a work-around to be able to display (within the mouse-over-help) the "\<empty\>" instead of just a plain \'\' (empty string) - "\<empty\>" being hopefully more readable and understandable to the user.